Dockerfile

```
@@ -75,40 +75,184 @@ ADD hack/scripts/ /usr/local/bin/
 
 COPY --from=docker /usr/local/bin/docker /usr/local/bin/dockerd /usr/local/bin/
 
-FROM ghcr.io/actions-runner-controller/actions-runner-controller/actions-runner-dind:ubuntu-22.04 AS build-container-ghaction
+# renovate: datasource=github-releases extractVersion=^gha-runner-scale-set-(?<version>.*)$ depName=actions/actions-runner-controller
+ARG ACTIONS_RUNNER_CONTROLLER_VERSION=0.10.1
+FROM scratch AS actions-runner-controller-source
+ADD https://github.com/actions/actions-runner-controller.git#${ACTIONS_RUNNER_CONTROLLER_VERSION}:runner /
+
+# Ref: https://github.com/actions/actions-runner-controller/blob/master/runner/actions-runner-dind.ubuntu-22.04.dockerfile
+FROM ubuntu:25.04 AS build-container-ghaction
+
+ARG TARGETPLATFORM
+# renovate: datasource=github-releases extractVersion=^v(?<version>.*)$ depName=actions/runner
+ARG RUNNER_VERSION=2.322.0
+# renovate: datasource=github-releases extractVersion=^v(?<version>.*)$ depName=actions/runner-container-hooks
+ARG RUNNER_CONTAINER_HOOKS_VERSION=0.6.2
+# Docker and Docker Compose arguments
+ARG CHANNEL=stable
+# renovate: datasource=github-releases extractVersion=^v(?<version>.*)$ depName=moby/moby
+ARG DOCKER_VERSION=27.5.1
+# renovate: datasource=github-releases depName=docker/compose
+ARG DOCKER_COMPOSE_VERSION=v2.33.0
+# renovate: datasource=github-releases extractVersion=^v(?<version>.*)$ depName=yelp/dumb-init
+ARG DUMB_INIT_VERSION=1.2.5
+ARG RUNNER_USER_UID=1001
+ARG DOCKER_GROUP_GID=121
+
 # renovate: datasource=github-releases depName=google/go-containerregistry
-ARG CRANE_VERSION=v0.20.2
+ARG CRANE_VERSION=v0.20.3
 # renovate: datasource=github-releases depName=mikefarah/yq
-ARG YQ_VERSION=v4.44.3
+ARG YQ_VERSION=v4.45.1
 # renovate: datasource=github-releases depName=getsops/sops
-ARG SOPS_VERSION=v3.9.1
+ARG SOPS_VERSION=v3.9.4
 # renovate: datasource=github-tags depName=aws/aws-cli
-ARG AWSCLI_VERSION=2.19.1
-USER root
-RUN apt update && \
-	apt upgrade -y && \
-	apt install -y \
-	--no-install-recommends \
-	curl \
-	diffoscope \
-	docker.io \
-	gh \
-	iproute2 \
-	iptables \
-	make \
-	mkisofs \
-	openssh-client \
-	ovmf \
-	qemu-system \
-	qemu-utils \
-	socat \
-	swtpm \
-	tmux \
-	unzip \
-	zstd
+ARG AWSCLI_VERSION=2.24.5
+
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update -y \
+    && apt-get install -y software-properties-common \
+    && add-apt-repository -y ppa:git-core/ppa \
+    && apt-get update -y \
+    && apt-get install -y --no-install-recommends \
+    curl \
+    ca-certificates \
+    git \
+    iptables \
+    jq \
+    software-properties-common \
+    sudo \
+    unzip \
+    zip \
+    && rm -rf /var/lib/apt/lists/*
```

Maybe removing cache could optimize (alternative: use cache mounts for lists and package cache)
```suggestion
    && rm -rf /var/lib/apt/lists/* && apt-get clean
```